### DISCUSSION TITLE
Getting SQLSTATE[HY000]: General error: 1419 error at end of the PimCore demo version installation

### DISCUSSION THREAD

**Discussion by tapli** [pimcore/pimcore#14682] [Q&A / Support] [answered]
I have installed pimcore-demo version on Windows using WSL, Apache2 and Mysql. The installation was successful. When I access the /admin page, it prompts with user id and password. Despite entering the correct user id and password, it says "Login failed". The page produces no background image/stylesheet. 
I have re-installed the application again and ended up having the same issue. The Pimcore version is 10.5.16. 

What am I missing?

**Comment by podarcis** [comment, marked as answer]
> Admin portal login failed

I suggest you first check your logs (`var/log/dev.log`). You might see the cause of the issue there.

If not, try to re-run `composer install` and check the output. No errors should occur and asset symlinks should have been updated. Then try again and check whether stylesheet/images show up. If not, check your webserver config (are Apache's .htaccess or Nginx rules in place?). If so and you still cannot login, try resetting your admin's password with console command `pimcore:user:reset-password` to something simple.

**Reply by tapli** [reply]
@podarcis 

Thanks for the reply. I checked the users table and did not find "admin" user. I believe PimCore is creating "admin" user at the end of the installation. And in my case, it failed with MySQL error. Let me re-install and post the MySQL error message.

**Reply by tapli** [reply]
@podarcis 

I have re-installed PimCore and this is the MySQL error message it throws at the end of the installation. Is this a known issue?

 * An exception occurred while executing 'DROP FUNCTION IF EXISTS PLUGIN_CMF_COLLECT_ASSET_SEGMENT_ASSIGNMENTS;

CREATE FUNCTION PLUGIN_CMF_COLLECT_ASSET_SEGMENT_ASSIGNMENTS(elementIdent INT) RETURNS TEXT
  READS SQL DATA
  BEGIN
    DECLARE segmentIds TEXT;
    DECLARE elementExists TINYINT;
    DECLARE breaks TINYINT;

    SELECT `segments` FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'asset' INTO segmentIds;
    SELECT COUNT(*) FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'asset' INTO elementExists;
    SELECT `breaksInheritance` FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'asset' INTO breaks;

    WHILE (elementExists = 0 OR breaks IS NULL OR breaks <> 1) AND elementIdent > 1 DO
      SELECT `parentId`  FROM `assets` WHERE `id` = elementIdent INTO elementIdent;
      SELECT CONCAT_WS(',', segmentIds, (SELECT `segments` FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'asset')) INTO segmentIds;
      SELECT COUNT(*) FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'asset' INTO elementExists;
      SELECT `breaksInheritance` INTO breaks FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'asset';
    END WHILE;

    RETURN segmentIds;
  END;
':

SQLSTATE[HY000]: General error: 1419 You do not have the SUPER privilege and binary logging is enabled (you *might* want to use the less safe log_bin_trust_function_creators variable)

**Reply by tapli** [reply]
@podarcis 

To resolve this issue, I updated the MySQL flag log_bin_trust_function_creators and set it to one. Then re-installed the application and it worked.

Thank you.

**Reply by podarcis** [reply]
Glad you solved it 👍
